test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -98,19 +97,14 @@ jruby-config :- jruby-schemas/JRubyConfig
          use-ssl?             (true? (:use-ssl options))
          http-client-settings (get-http-client-settings
                                (select-keys options [:ssl-protocols :cipher-suites]))
-         scripting-container (-> (ScriptingContainer. LocalContextScope/SINGLETHREAD
-                                                      LocalVariableBehavior/PERSISTENT)
-                                 (jruby-internal/init-jruby jruby-config))]
-     (doto scripting-container
-       (.setEnvironment (jruby-core/managed-environment (jruby-core/get-system-env) jruby-config))
-       (.setLoadPaths (jruby-puppet-core/managed-load-path jruby-puppet-testutils/ruby-load-path))
-       (.runScriptlet "require 'jar-dependencies'")
-       (.runScriptlet "require 'puppet/server/http_client'"))
-
+         scripting-container (-> (jruby-impl-core/create-pool jruby-config)
+                                 (jruby-core/borrow-from-pool :http-client-test [])
+                                 :scripting-container)]
+     (.runScriptlet scripting-container "require 'puppet/server/http_client'")
      (let [http-client-class (.runScriptlet scripting-container "Puppet::Server::HttpClient")]
        (.callMethod scripting-container http-client-class "initialize_settings" http-client-settings Object))
      (.runScriptlet scripting-container (str "require 'puppet/server/http_client';"
-                                             (format "c = Puppet::Server::HttpClient.new('localhost', %d, {:use_ssl => %s});"
+                                             (format "$c = Puppet::Server::HttpClient.new('localhost', %d, {:use_ssl => %s});"
```

Comment:
Yeah, probably would be better to change these tests to just get the `HttpClient` object back from the container and make calls directly on that.  Just took the lazy way out to avoid thinking about how to do all of the right `.callMethodWithArgArray` calls and interop for the different arguments.